nat-lab/tests/test_pq.py

```diff
@@ -372,3 +378,108 @@ async def test_dns_with_pq(
 
         # Expect this to work as well after the secret key change
         await query_dns(client_conn, "google.com")
+
+
+@pytest.mark.parametrize(
+    "setup",
+    [
+        pytest.param(
+            SetupParameters(
+                connection_tag=ConnectionTag.DOCKER_CONE_CLIENT_1,
+                adapter_type_override=TelioAdapterType.NEP_TUN,
+                connection_tracker_config=generate_connection_tracker_config(
+                    ConnectionTag.DOCKER_CONE_CLIENT_1,
+                    nlx_1_limits=ConnectionLimits(1, 2),
+                ),
+                is_meshnet=False,
+            ),
+        ),
+    ],
+)
+async def test_pq_vpn_silent_pq_upgrader(
+    setup: SetupParameters,
+) -> None:
+    async with AsyncExitStack() as exit_stack:
+        setup.features.post_quantum_vpn.handshake_retry_interval_s = 1
+
+        env = await exit_stack.enter_async_context(
+            setup_environment(exit_stack, [setup])
```

Comment:
Same us: https://github.com/NordSecurity/libtelio/pull/991/files#r1863529347

Reply:
Done. Waiting for CI to finish